libs/deso-protocol/src/lib/utils/Utils.ts

```diff
@@ -186,3 +207,373 @@ export const publicKeyToDeSoPublicKey = (
   const key = publicKey.getPublic().encode('array', true);
   return bs58check.encode(Buffer.from([...prefix, ...key]));
 };
+
+export function seedHexToECKeyPair(seedHex: string): EC.KeyPair {
+  const ec = new EC('secp256k1');
+  return ec.keyFromPrivate(seedHex);
+}
+
+export const seedHexToPrivateKey = (seedHex: string): EC.KeyPair => {
+  const ec = new EC('secp256k1');
+  return ec.keyFromPrivate(seedHex);
+};
+
+export const signTransaction = (
+  seedHex: string,
+  transactionHex: string,
+  isDerivedKey: boolean
+): string => {
+  const privateKey = seedHexToPrivateKey(seedHex);
+
+  const transactionBytes = new Buffer(transactionHex, 'hex');
+  const transactionHash = new Buffer(sha256.x2(transactionBytes), 'hex');
+  const signature = privateKey.sign(transactionHash, { canonical: true });
+  const signatureBytes = new Buffer(signature.toDER());
+  const signatureLength = uvarint64ToBuf(signatureBytes.length);
+
+  // If transaction is signed with a derived key, use DeSo-DER recoverable signature encoding.
+  if (isDerivedKey) {
+    signatureBytes[0] += 1 + (signature.recoveryParam as number);
+  }
+
+  const signedTransactionBytes = Buffer.concat([
+    // This slice is bad. We need to remove the existing signature length field prior to appending the new one.
+    // Once we have frontend transaction construction we won't need to do this.
+    transactionBytes.slice(0, -1),
+    signatureLength,
+    signatureBytes,
+  ]);
+
+  return signedTransactionBytes.toString('hex');
+};
+
+export const encryptMessage = async (
+  deso: Deso,
+  messageToSend: string,
+  derivedKeyResponse: Partial<DerivedPrivateUserInfo>,
```

Comment:
I'd prefer not to pass the `derivedKeyResponse` since that limits the utility of this function, can we pass in the values we need from it instead? and will probably need an `isDerived` flag to support the submitTransaction piece